### PR TITLE
Adds distance matrix to neighborhood; provides another function

### PR DESCRIPTION
**Pull request by sbromberger**
to return a tuple of (vector, distance)

### CODE REVIEW

**Comment by yeesian** [comment]
do you need tests for the new `egonet()` function with non-standard weights? otherwise LGTM

**Comment by sbromberger** [comment]
> do you need tests for the new egonet() function with non-standard weights? otherwise LGTM

Oh, good catch. Thank you.

File: src/connectivity.jl [hunk 1]
```diff
@@ -305,43 +305,52 @@ function attracting_components end
...
+                if vdist <= d
+                    push!(Q, vertex)
+                    push!(neighs, vertex)
+                    dists[vertex] = vdist
+                end
             end
         end
     end
-    return neighs
+    return [(x::T, dists[x]::U) for x in neighs]
```
Comment (jpfairbanks) [inline src/connectivity.jl]: This is a workaround for the issue https://github.com/JuliaLang/julia/issues/24971? you prefer the type assert over the `let` workaround?

File: src/connectivity.jl [hunk 2]
```diff
@@ -305,43 +305,52 @@ function attracting_components end
...
+    first.(neighborhood_with_distances(g, v, d, distmx; dir=dir))
+
+neighborhood_with_distances(g::AbstractGraph{T}, v::Integer, d, distmx::AbstractMatrix{U}=weights(g); dir=:out) where T<:Integer where U<:Real =
+    (dir == :out) ? _neighborhood(g, v, d, distmx, out_neighbors) : _neighborhood(g, v, d, distmx, in_neighbors)
 
-function _neighborhood(g::AbstractGraph{T}, v::Integer, d::Integer, neighborfn::Function) where T
-    d < 0 && return Vector{T}()
+
+function _neighborhood(g::AbstractGraph{T}, v::Integer, d::Real, distmx::AbstractMatrix{U}, neighborfn::Function) where T<:Integer where U<:Real
+    d < 0 && return Vector{Tuple{T, U}}()
```
Comment (jpfairbanks) [inline src/connectivity.jl]: Nice a null result rather than `error` i like it.

File: src/connectivity.jl [hunk 3]
```diff
@@ -305,43 +305,52 @@ function attracting_components end
...
 ### Optional Arguments
 - `dir=:out`: If `g` is directed, this argument specifies the edge direction
 with respect to `v` of the edges to be considered. Possible values: `:in` or `:out`.
 """
-neighborhood(g::AbstractGraph, v::Integer, d::Integer; dir=:out) = (dir == :out) ?
-    _neighborhood(g, v, d, out_neighbors) : _neighborhood(g, v, d, in_neighbors)
+neighborhood(g::AbstractGraph{T}, v::Integer, d, distmx::AbstractMatrix{U}=weights(g); dir=:out) where T<:Integer where U<:Real =
+    first.(neighborhood_with_distances(g, v, d, distmx; dir=dir))
+
+neighborhood_with_distances(g::AbstractGraph{T}, v::Integer, d, distmx::AbstractMatrix{U}=weights(g); dir=:out) where T<:Integer where U<:Real =
```
Comment (jpfairbanks) [inline src/connectivity.jl]: this name is super long and has 2 _. Can anyone come up with something better? dneighborhood?  dihood?

**Comment by jpfairbanks** [comment]
I put some line comments in, my big thing is the name. so long. so many underscores

File: src/connectivity.jl (hunk 1 above)
Comment (sbromberger) [reply src/connectivity.jl]: I don't necessarily like the type assert over `let`, but I came up with this independently of the solution in that issue. Is there a reason to prefer one over the other?

File: src/connectivity.jl (hunk 2 above)
Comment (sbromberger) [reply src/connectivity.jl]: It makes sense, right? Nothing in your negative neighborhood.

File: src/connectivity.jl (hunk 3 above)
Comment (sbromberger) [reply src/connectivity.jl]: I don't like it either.

**Comment by jpfairbanks** [comment]
@yeesian do you have a better name than `neighborhood_with_distances`?

**Comment by yeesian** [comment]
Maybe `neighborhood_dists()`?

File: src/connectivity.jl [hunk 4]
```diff
@@ -305,43 +305,52 @@ function attracting_components end
 end
 
 """
-    neighborhood(g, v, d)
+    neighborhood(g, v, d, distmx=weights(g))
+    neighborhood_with_distances(g, v, d, distmx=weights(g))
 
-Return a vector of the vertices in `g` at a geodesic distance less or equal to `d`
-from `v`.
+Return a vector of each vertex in `g` at a geodesic distance less than or equal to `d`, where distances
```
Comment (yeesian) [inline src/connectivity.jl]: why is it called a geodesic distance?

**Comment by sbromberger** [comment]
I liked `neighborhood_dists` better.

**Comment by yeesian** [comment]
I don't have any strong opinion on the matter

**Comment by sbromberger** [comment]
@jpfairbanks - `neighborhood_dists`?

**Comment by sbromberger** [comment]
@jpfairbanks we good to go?

**Comment by jpfairbanks** [comment]
yup neighborhood_dists is perfect.

**Comment by sbromberger** [comment]
@jpfairbanks - mind approving?

**Comment by sbromberger** [comment]
Holding this one open to test @jpfairbanks ' approval permissions. I dislike GitHub permissions. 👎

**Comment by sbromberger** [comment]
Cool - you can approve. Can you test to make sure you can merge as well (once the retest passes)?

**Comment by sbromberger** [comment]
@jpfairbanks' first merge! w00t.

**Pull request merged**